pkg/exemplar/exemplar.go

```diff
@@ -13,12 +13,27 @@
 
 package exemplar
 
-import "github.com/prometheus/prometheus/pkg/labels"
+import (
+	"github.com/prometheus/prometheus/pkg/labels"
+)
 
 // Exemplar is additional information associated with a time series.
 type Exemplar struct {
-	Labels labels.Labels
-	Value  float64
-	HasTs  bool
-	Ts     int64
+	Labels labels.Labels `json:"labels"`
+	Value  float64       `json:"value"`
+	Ts     int64         `json:"timestamp"`
+	HasTs  bool          `json:"hasTimestamp"`
+}
+
+// Equals compares if the exemplar e is the same as e2.
+func (e Exemplar) Equals(e2 Exemplar) bool {
+	if e.Labels.Hash() != e2.Labels.Hash() {
```

Comment:
I don't know how often this is called (or how performant it needs to be), but iterating through the labels and comparing would be faster as it would allow the code to bail out on the first mismatch and doesn't require allocating additional memory.

If extreme performance is not a consideration I agree that the current code is more readable.